src/argus/filter/queryset_filters.py

```diff
@@ -0,0 +1,128 @@
+from argus.filter.filterwrapper import FilterWrapper, FilterBlobType
+from argus.incident.models import Incident
+from argus.notificationprofile.models import Filter
+from argus.notificationprofile.models import NotificationProfile
+
+
+__all__ = ["QuerySetFilter"]
+
+
+def _incidents_with_source_systems(incident_queryset, filterblob: FilterBlobType):
+    source_list = filterblob.get("sourceSystemIds", [])
+    if source_list:
+        return incident_queryset.filter(source__in=source_list).distinct()
+    return incident_queryset.distinct()
+
+
+def _incidents_with_tags(incident_queryset, filterblob: FilterBlobType):
+    tags_list = filterblob.get("tags", [])
+    if tags_list:
+        return incident_queryset.from_tags(*tags_list)
+    return incident_queryset.distinct()
+
+
+def _incidents_fitting_tristates(incident_queryset, filterblob: FilterBlobType):
+    fitting_incidents = incident_queryset
+    filter_open = filterblob.get("open", None)
+    filter_acked = filterblob.get("acked", None)
+    filter_stateful = filterblob.get("stateful", None)
+
+    if filter_open is True:
+        fitting_incidents = fitting_incidents.open()
+    if filter_open is False:
+        fitting_incidents = fitting_incidents.closed()
+    if filter_acked is True:
+        fitting_incidents = fitting_incidents.acked()
+    if filter_acked is False:
+        fitting_incidents = fitting_incidents.not_acked()
+    if filter_stateful is True:
+        fitting_incidents = fitting_incidents.stateful()
+    if filter_stateful is False:
+        fitting_incidents = fitting_incidents.stateless()
+    return fitting_incidents.distinct()
+
+
+def _incidents_fitting_maxlevel(incident_queryset, filterblob: FilterBlobType):
+    maxlevel = filterblob.get("maxlevel", None)
+    if not maxlevel:
+        return incident_queryset.distinct()
+    return incident_queryset.filter(level__lte=maxlevel).distinct()
+
+
+def _incidents_fitting_event_types(incident_queryset, filterblob: FilterBlobType):
+    event_types = filterblob.get("event_types", [])
+    if not event_types:
+        return incident_queryset.distinct()
+    return incident_queryset.prefetch_related("events").filter(events__type__in=event_types).distinct()
+
+
+class QuerySetFilter:
+    @staticmethod
+    def filtered_incidents(filterblob: FilterBlobType, incident_queryset=None):
+        if not incident_queryset:
```

Comment:
```suggestion
        if incident_queryset is None:
```
otherwise this will also be true if for some reason the given query set is empty